Website/src/rendering/src/components/Forms/AddressBookForm.tsx

```diff
@@ -0,0 +1,249 @@
+import { DBuyerAddress } from '../../models/ordercloud/DBuyerAddress';
+import { FormEvent, useEffect, useState } from 'react';
+import { GeographyService } from '../../services/GeographyService';
+import Spinner from '../../components/ShopCommon/Spinner';
+import Link from 'next/link';
+
+type AddressBookFormProps = {
+  address?: DBuyerAddress;
+  onSubmit?: (address: DBuyerAddress) => void;
+  isEditing?: boolean;
+  onCancelEdit?: () => void;
+  loading?: boolean;
+  prefix?: string; // needed when more that one form on checkout page
+};
+
+const AddressBookForm = (props: AddressBookFormProps): JSX.Element => {
+  /**
+   * TODO:
+   * 1. Add better postal code validation based on country selected by using masks (react-input-mask)
+   * 2. Add more supported countries, currently only support US & Canada
+   * 3. Disable submit button unless there are actual changes
+   * 4. Add ability to discard changes (this would only be on editing addresses)
+   * 5. Remove mocked address once saved addresses are a thing
+   */
+  const countries = GeographyService.getCountries();
+  const [states, setStates] = useState(
+    GeographyService.getStatesOrProvinces(props.address?.Country || countries[0].code)
+  );
+  const [addressName, setAddressName] = useState(props?.address?.AddressName || '');
+  const [country, setCountry] = useState(props?.address?.Country || '');
+  const [firstName, setFirstName] = useState(props?.address?.FirstName || '');
+  const [lastName, setLastName] = useState(props?.address?.LastName || '');
+  const [street1, setStreet1] = useState(props.address?.Street1 || '');
+  const [street2, setStreet2] = useState(props.address?.Street2 || '');
+  const [city, setCity] = useState(props?.address?.City || '');
+  const [state, setState] = useState(props?.address?.State || '');
+  const [zip, setZip] = useState(props?.address?.Zip || '');
+  const [defaultBilling, setDefaultBilling] = useState(props?.address?.xp?.DefaultBilling);
```

Comment:
By saving the "default" properties in the address extended properties (xp), it allows multiple addresses to all be the same default:
![image](https://user-images.githubusercontent.com/2000200/171631535-b14897a8-1842-4bb0-a16f-30e85c3d4808.png)

We should store those defaults in the user object instead of the address object. We could store the ID of the default shipping address, and the ID of the default billing address. That way, only one address at a time could be the default of its type.